gix-fs/tests/stack/mod.rs

```diff
@@ -35,97 +35,97 @@ fn p(s: &str) -> &Path {
 
     assert_eq!(
-        p("\\\\?\\base").join(absolute),
-        p("\\\\?\\base\\absolute"),
+        p(r"\\?\base").join(looks_absolute),
+        p(r"\\?\base\absolute"),
         "absolute1 + unix-absolute2 = joined result with backslash"
     );
     assert_eq!(
-        p("\\\\.\\base").join(absolute),
-        p("\\\\.\\base\\absolute"),
+        p(r"\\.\base").join(looks_absolute),
+        p(r"\\.\base\absolute"),
         "absolute1 + absolute2 = joined result with backslash (device namespace)"
     );
     assert_eq!(
-        p("\\\\?\\base").join(bs_absolute),
-        p("\\\\?\\base\\absolute"),
+        p(r"\\?\base").join(bs_looks_absolute),
+        p(r"\\?\base\absolute"),
         "absolute1 + absolute2 = joined result"
     );
     assert_eq!(
-        p("\\\\.\\base").join(bs_absolute),
-        p("\\\\.\\base\\absolute"),
+        p(r"\\.\base").join(bs_looks_absolute),
+        p(r"\\.\base\absolute"),
         "absolute1 + absolute2 = joined result (device namespace)"
     );
 
     assert_eq!(p("/").join("C:"), p("C:"), "unix-absolute + win-drive = win-drive");
     assert_eq!(
         p("d:/").join("C:"),
         p("C:"),
-        "d-drive + c-drive = c-drive - interesting, as C: is supposed to be relative"
+        "d-drive + c-drive-relative = c-drive-relative - C: is relative but not on D:"
     );
     assert_eq!(
-        p("d:\\").join("C:\\"),
-        p("C:\\"),
+        p(r"d:\").join(r"C:\"),
+        p(r"C:\"),
         "d-drive-with-bs + c-drive-with-bs = c-drive-with-bs - nothing special happens with backslashes"
     );
     assert_eq!(
-        p("c:\\").join("\\\\.\\"),
-        p("\\\\.\\"),
+        p(r"c:\").join(r"\\.\"),
+        p(r"\\.\"),
         "c-drive-with-bs + device-namespace-unc = device-namespace-unc"
     );
     assert_eq!(
         p("/").join("C:/"),
-        p("C:\\"),
+        p(r"C:\"),
         "unix-absolute + win-drive = win-drive, strangely enough it changed the trailing slash to backslash, so better not have trailing slashes"
     );
-    assert_eq!(p("/").join("C:\\"), p("C:\\"), "unix-absolute + win-drive = win-drive");
+    assert_eq!(p("/").join(r"C:\"), p(r"C:\"), "unix-absolute + win-drive = win-drive");
     assert_eq!(
-        p("\\\\.").join("C:"),
+        p(r"\\.").join("C:"),
         p("C:"),
-        "device-namespace-unc + win-drive-relative = win-drive-relative - c: was supposed to be relative, but it's not acting like it."
+        r"device-namespace-unc + win-drive-relative = win-drive-relative - C: as a relative path is not the C: device, so this is not \\.\C:"
     );
     assert_eq!(p("relative").join("C:"), p("C:"), "relative + win-drive = win-drive");
 
     assert_eq!(
-        p("/").join("\\\\localhost"),
-        p("\\localhost"),
-        "unix-absolute + win-absolute-unc = win-absolute-unc"
+        p("/").join(r"\\localhost"),
+        p(r"\localhost"),
+        "unix-absolute + win-absolute-unc-host = strangely, single-backslashed host"
```

Comment:
This strange, and my puzzlement, relates to https://github.com/Byron/gitoxide/pull/1374#discussion_r1721079809.